index.js

```diff
@@ -18,48 +18,70 @@ StylusFilter.prototype.constructor = StylusFilter;
 StylusFilter.prototype.extensions = ['styl'];
 StylusFilter.prototype.targetExtension = 'css';
 
-StylusFilter.prototype.processString = function (str) {
+StylusFilter.prototype.processString = function(str) {
 	var opts = this.options;
+
+	var importRegex = new RegExp(/@import ['"].*['"]/);
+	while (importRegex.test(str)) {
+		var imp = importRegex.exec(str);
+		var sourceReg = new RegExp(/['"].*['"]/);
+		var extReg = new RegExp(/.styl/);
+		var src = sourceReg.exec(imp[0])[0].replace(/"/g, '').replace(/'/g, '');
+		if (!extReg.test(src)) {
+			src = src + '.styl';
+		}
+
+		src = this.inputTree.inputTree.inputTree + '/' + this.inputTree.srcDir + '/' + src;
```

Comment:
If you'll remove this line of code you'll get 'failed to locate @import file...'
That's the way I found for building the full path of the file (I'm using [broccoli-funnel]https://github.com/broccolijs/broccoli-funnel)
If you have more elegant way to get the path of the import file I'll be happy to learn.
Thanks.